clis/teliod/src/qnap.rs

```
@@ -0,0 +1,458 @@
+            fs::remove_file(PID_FILE)
+        }
+        Err(e) => Err(e),
+    }
+}
+
+fn start_daemon() -> Response {
+    if is_teliod_running() {
+        return text_response(StatusCode::BAD_REQUEST, "Application is already running.");
+    }
+
+    let mut teliod_log_file = match fs::OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .open(TELIOD_LOG)
+    {
+        Ok(file) => file,
+        Err(_) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                "Failed to open teliod log file.",
+            );
+        }
+    };
+    let stdout = match teliod_log_file.try_clone() {
+        Ok(file) => Stdio::from(file),
+        Err(error) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to start the application: {error}"),
+            )
+        }
+    };
+    let stderr = match teliod_log_file.try_clone() {
+        Ok(file) => Stdio::from(file),
+        Err(error) => {
+            return text_response(
+                StatusCode::INTERNAL_SERVER_ERROR,
+                format!("Failed to start the application: {error}"),
+            )
+        }
+    };
+    match Command::new("setsid")
+        .arg(TELIOD_BIN)
+        .arg("daemon")
+        .arg(TELIOD_CFG)
+        .stdout(stdout)
+        .stderr(stderr)
+        .spawn()
+    {
+        Ok(process) => {
+            let _ = teliod_log_file.write_all(format!("Process ID: {}\n", process.id()).as_bytes());
+            text_response(StatusCode::CREATED, "Application started successfully.")
+        }
+        Err(error) => text_response(
+            StatusCode::INTERNAL_SERVER_ERROR,
+            format!("Failed to start the application: {error}"),
+        ),
+    }
+}
+
+fn stop_daemon() -> Response {
+    if shutdown_teliod().is_ok() {
+        text_response(StatusCode::OK, "Application stopped successfully.")
+    } else {
```

Let's not ignore this error and include it in below messages or log it somehow.

---

Added shutdown error log